configure: error: OpenSSL is not installed

Running `pyenv doctor` on a Mac with homebrew for Apple Silicon, I get the following error about OpenSSL not being installed:

```
❯ pyenv doctor
Cloning /Users/chris/.pyenv/plugins/pyenv-doctor/bin/.....
Installing python-pyenv-doctor...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 12.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/k4/9gpkctsn0dvdwmjyylmdzb5h0000gn/T/python-build.20220105233748.89082
Results logged to /var/folders/k4/9gpkctsn0dvdwmjyylmdzb5h0000gn/T/python-build.20220105233748.89082.log

Last 10 log lines:
checking readline/readline.h, usability... no
checking readline/readline.h, presence... no
checking for readline/readline.h,... no
checking readline/rlconf.h usability... yes
checking readline/rlconf.h presence... yes
checking for readline/rlconf.h... yes
checking for SSL_library_init in -lssl... no
configure: WARNING: OpenSSL <1.1 not installed. Checking v1.1 or beyond...
checking for OPENSSL_init_ssl in -lssl... no
configure: error: OpenSSL is not installed.
Problem(s) detected while checking system.

See https://github.com/pyenv/pyenv/wiki/Common-build-problems for known solutions.
```

I think the issue is because the paths to homebrew openssl are being included correctly.  Is there a reason why the homebrew path logic can't be simplified into something like this?

```diff
diff --git a/bin/pyenv-doctor b/bin/pyenv-doctor
index d3b0188..cfbea42 100755
--- a/bin/pyenv-doctor
+++ b/bin/pyenv-doctor
@@ -155,16 +155,10 @@ case "$(uname -s)" in
   ## to override system libraries installed by Homebrew,
   ## we must explicitly specify the library path in "-L".
   if command -v brew 1>/dev/null 2>&1; then # Homebrew
-    # According to https://docs.brew.sh/Installation,
-    # "/usr/local" is for Intel binaries,
-    # "/opt/homebrew" for ARM.
-    if [ "$(uname -m)" = arm64 ]; then
-      [ -d "/opt/homebrew/include" ] && export CPPFLAGS="-I/opt/homebrew/include $CPPFLAGS"
-      [ -d "/opt/homebrew/lib" ] && export LDFLAGS="-L/opt/homebrew/lib $LDFLAGS"
-    else
-      [ -d "/usr/local/include" ] && export CPPFLAGS="-I/usr/local/include $CPPFLAGS"
-      [ -d "/usr/local/lib" ] && export LDFLAGS="-L/usr/local/lib $LDFLAGS"
-    fi
+    [ -d "$(brew --prefix)/include" ] && export CPPFLAGS="-I$(brew --prefix)/include $CPPFLAGS"
+    [ -d "$(brew --prefix openssl)/include" ] && export CPPFLAGS="-I$(brew --prefix openssl)/include $CPPFLAGS"
+    [ -d "$(brew --prefix)/lib" ] && export LDFLAGS="-L$(brew --prefix)/lib $LDFLAGS"
+    [ -d "$(brew --prefix openssl)/lib" ] && export LDFLAGS="-L$(brew --prefix openssl)/lib $LDFLAGS"
   fi
   ;;
 esac
```

--- a/bin/pyenv-doctor
+++ b/bin/pyenv-doctor
@@ -155,16 +155,10 @@ case "$(uname -s)" in
   ## to override system libraries installed by Homebrew,
   ## we must explicitly specify the library path in "-L".
   if command -v brew 1>/dev/null 2>&1; then # Homebrew
-    # According to https://docs.brew.sh/Installation,
-    # "/usr/local" is for Intel binaries,
-    # "/opt/homebrew" for ARM.
-    if [ "$(uname -m)" = arm64 ]; then
-      [ -d "/opt/homebrew/include" ] && export CPPFLAGS="-I/opt/homebrew/include $CPPFLAGS"
-      [ -d "/opt/homebrew/lib" ] && export LDFLAGS="-L/opt/homebrew/lib $LDFLAGS"
-    else
-      [ -d "/usr/local/include" ] && export CPPFLAGS="-I/usr/local/include $CPPFLAGS"
-      [ -d "/usr/local/lib" ] && export LDFLAGS="-L/usr/local/lib $LDFLAGS"
-    fi
+    [ -d "$(brew --prefix)/include" ] && export CPPFLAGS="-I$(brew --prefix)/include $CPPFLAGS"
+    [ -d "$(brew --prefix openssl)/include" ] && export CPPFLAGS="-I$(brew --prefix openssl)/include $CPPFLAGS"
+    [ -d "$(brew --prefix)/lib" ] && export LDFLAGS="-L$(brew --prefix)/lib $LDFLAGS"
+    [ -d "$(brew --prefix openssl)/lib" ] && export LDFLAGS="-L$(brew --prefix openssl)/lib $LDFLAGS"
   fi
   ;;
 esac

